### ISSUE TITLE
bug: Optional is not reflected in docs

### ISSUE THREAD

**Issue by ollydixon** [jlalmes/trpc-openapi#217]
I searched previous issues but couldn't find anything like this.

When using the input

```
.input(
    z.object({
      nameOfUser: z.string().optional(),
    }),
  )
```

I get the following:

<img width="255" alt="Screenshot 2022-12-30 at 15 57 27" src="https://user-images.githubusercontent.com/6940726/210084117-3afd371f-8f47-4037-b69a-9dd7cbca8715.png">

This is incorrect, 

It should be 

```
{
  "nameOfUser": "string?"
}
```

**Comment by jlalmes** [comment]
Your screenshot is showing an "Example Value", which is correct. If you press "Schema" you will see as follows: 

<img width="292" alt="Screenshot 2022-12-31 at 18 15 22" src="https://user-images.githubusercontent.com/69924001/210152328-e0dc248d-08c5-456e-8f6b-3aa2a0a52a5d.png">

**Comment by ollydixon** [comment]
@jlalmes ah ok thanks, that's not good. It will leave developers on the 'believe' that nothing is optional.

**Comment by ollydixon** [comment]
Also it doesn't actually show as optional here?

<img width="317" alt="Screenshot 2022-12-31 at 19 17 50" src="https://user-images.githubusercontent.com/6940726
/210152379-f2a9d6ec-5e72-4502-aa67-1e07f13a3b62.png">

**Comment by ollydixon** [comment]
Also it doesn't show as optional here?

<img width="335" alt="Screenshot 2022-12-31 at 19 18 23" src="https://user-images.githubusercontent.com/6940726/210152393-e4b2aab4-b06e-468e-b68c-0c5baf9ec103.png">

**Comment by jlalmes** [comment]
> It will leave developers on the 'believe' that nothing is optional.

This is an issue with [swagger-ui](https://swagger.io/tools/swagger-ui/). Please open an issue with them or use an alternative OpenAPI documentation viewer.

> Also it doesn't show as optional here?

Can you share a reproduction.

**Comment by ollydixon** [comment]
@jlalmes great, thanks for helping

```OpenAPI documentation viewer```
Ahh, can I use alternatives with this library, how can I do that?

It's the same as before

```
export const adminCreateJwt = publicProcedure
  .meta({ openapi: { method: 'POST', path: '/adminCreateJwt', tags: ['admin'], protect: false, description: 'Optional name of user' } })
  .input(
    z.object({
      nameOfUser: z.string().optional(),
    }),
  )
   // We have to label the output each time now. It's not inferred like TS-playground.
  .output(
    z.object({
      jwt: z.string(),
    }),
  )
  .mutation(async ({ ctx, input }) => {
    if (isDev() || isDevLive()) {
      ... some code
    }

    return {
      jwt: 'live',
    };
  });
```

And the setup

```
import { generateOpenApiDocument } from 'trpc-openapi';
import { appRouter } from './routes';

export const openApiDocument = generateOpenApiDocument(appRouter, {
  title: 'tRPC OpenAPI',
  description: 'Vegiano API',
  version: '1.0.0',
  baseUrl: `http://localhost:${process.env.PORT ?? 8080}/trpc`,
  tags: [
    'admin',
    'auth',
    //
  ],
});
```

**Comment by jlalmes** [comment]
My bad, the screenshot you shared is actually correct. It shows that the `nameOfUser` is optional because it does not have an asterisks after it.

What you will want to do is this:
```diff
.input(
  z.object({
    nameOfUser: z.string().optional(),
-  }),
+ }).optional(),
)
```

However I've just tested this and it still shows the request body is required because of a bug I've just found & fixed. Will push & publish this early next week.

Have a great new year!
